drivers/perf: add LRW DDR PMU support

GooTal opened the pull request:
fixed: #94

driver inclusion
category: feature
Link: https://github.com/RVCK-Project/rvck/issues/94

--------------------------------

This patch adds LRW DDR PMU support via the perf framework.

The implementation includes:
  - Registering a PMU driver for the LRW DDR controller
  - Providing basic counter support for DDR performance metrics
  - Integrating with the perf subsystem for user-space access

GooTal commented:
cc: @jay7275

oervci commented:
Kernel build success!

oervci commented:
kunit test done. log:https://jenkins.oerv.ac.cn/job/rvck-pipeline/job/kunit-test/186/consoleFull

wangliu-iscas commented:
/check

Commit:
drivers/perf: add LRW DDR PMU support

driver inclusion
category: feature
Link: https://github.com/RVCK-Project/rvck/issues/94

--------------------------------

This patch adds LRW DDR PMU support via the perf framework.

The implementation includes:
  - Registering a PMU driver for the LRW DDR controller
  - Providing basic counter support for DDR performance metrics
  - Integrating with the perf subsystem for user-space access

Signed-off-by: Jie Feng <feng.jie8@zte.com.cn>
Signed-off-by: Qingtao Liu <liu.qingtao2@zte.com.cn>

GooTal commented:
@sterling-teng 您好，已经rebase，验证完成
